cmd/oras/root/discover.go

```diff
@@ -136,25 +148,17 @@ func runDiscover(cmd *cobra.Command, opts *discoverOptions) error {
 	if err != nil {
 		return err
 	}
-	if handler.MultiLevelSupported() {
-		if err := fetchAllReferrers(ctx, repo, desc, opts.artifactType, handler); err != nil {
-			return err
-		}
-	} else {
-		refs, err := registry.Referrers(ctx, repo, desc, opts.artifactType)
-		if err != nil {
-			return err
-		}
-		for _, ref := range refs {
-			if err := handler.OnDiscovered(ref, desc); err != nil {
-				return err
-			}
-		}
+	if err := fetchAllReferrers(ctx, repo, desc, opts.artifactType, handler, opts.depth); err != nil {
+		return err
 	}
 	return handler.Render()
 }
 
-func fetchAllReferrers(ctx context.Context, repo oras.ReadOnlyGraphTarget, desc ocispec.Descriptor, artifactType string, handler metadata.DiscoverHandler) error {
+func fetchAllReferrers(ctx context.Context, repo oras.ReadOnlyGraphTarget, desc ocispec.Descriptor, artifactType string, handler metadata.DiscoverHandler, depth int) error {
+	var nextDepth int
+	if depth > 0 {
+		nextDepth = depth - 1
+	}
```

Comment:
nit: this part should be added just before line 167

Reply:
moved.